.github/workflows/ci.yml

```diff
@@ -0,0 +1,139 @@
+        uses: actions/cache@v2
+        id: cached-workspace
+        with:
+          path: |
+            _build/*/*/*
+            !_build/*/*/payments
+            ebin
+          key: workspace-${{ github.run_id }}
+      - name: Compile
+        run: rebar3 as test compile
+
+  xref:
+    needs: [compile]
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: "25.0"
+          rebar3-version: "3.18.0"
+      - name: Configure SSH key
+        uses: ./.github/actions/configure-ssh-key
+        with:
+          ssh_key: ${{ secrets.PRIVATE_SSH_KEY }}
+      - name: Fetch workspace from cache
+        uses: actions/cache@v2
+        id: cached-workspace
+        with:
+          path: |
+            _build/*/*/*
+            !_build/*/*/payments
+            ebin
+          key: workspace-${{ github.run_id }}
+      - name: Run xref
+        run: rebar3 xref
+
+  ct:
+    needs: [compile]
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: "25.0"
+          rebar3-version: "3.18.0"
+      - name: Configure SSH key
+        uses: ./.github/actions/configure-ssh-key
+        with:
+          ssh_key: ${{ secrets.PRIVATE_SSH_KEY }}
+      - name: Fetch workspace from cache
+        uses: actions/cache@v2
+        id: cached-workspace
+        with:
+          path: |
+            _build/*/*/*
+            !_build/*/*/payments
+            ebin
+          key: workspace-${{ github.run_id }}
+      - name: Run CT tests
+        run: rebar3 ct
+
+  eunit:
+    needs: [compile]
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: "25.0"
+          rebar3-version: "3.18.0"
+      - name: Configure SSH key
+        uses: ./.github/actions/configure-ssh-key
+        with:
+          ssh_key: ${{ secrets.PRIVATE_SSH_KEY }}
+      - name: Fetch workspace from cache
+        uses: actions/cache@v2
+        id: cached-workspace
+        with:
+          path: |
+           _build/*/*/*
+            !_build/*/*/payments
+            ebin
+          key: workspace-${{ github.run_id }}
+      - name: Run EUnit tests
+        run: rebar3 eunit
+
+  dialyzer:
+    needs: [compile]
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: "25.0"
+          rebar3-version: "3.18.0"
+      - name: Configure SSH key
+        uses: ./.github/actions/configure-ssh-key
+        with:
+          ssh_key: ${{ secrets.PRIVATE_SSH_KEY }}
+      - name: Fetch workspace from cache
+        uses: actions/cache@v2
+        id: cached-workspace
+        with:
+          path: |
+            _build/*/*/*
+            !_build/*/*/payments
+            ebin
+          key: workspace-${{ github.run_id }}
```

Comment:
Same as for the reps cache, OTP version and maybe rebar3 version. Also, it's nice if you could add the PLT to some cache. Will speed up the builds a lot.

It live in `_build/default/rebar3_25.0.3_plt`, so something like `_build/default/rebar3_*_plt` perhaps?